xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -725,6 +744,72 @@ ProcColorSpaces CProcessorHD::CalculateDXGIColorSpaces(const DXGIColorSpaceArgs&
   const bool limited = DX::Windowing()->UseLimitedColor();
   const bool supportHDR = DX::Windowing()->IsHDROutput() && (m_bSupportHDR10Limited || !limited);
 
-  return ProcColorSpaces{GetDXGIColorSpaceSource(csArgs, supportHDR, m_bSupportHLG),
-                         GetDXGIColorSpaceTarget(csArgs, supportHDR, limited)};
+  return ProcColorSpaces{
+      GetDXGIColorSpaceSource(csArgs, supportHDR, m_bSupportHLG, m_BT2020TopLeft),
+      GetDXGIColorSpaceTarget(csArgs, supportHDR, limited)};
+}
+
+bool CProcessorHD::IsBT2020Supported()
+{
+  ComPtr<ID3D11VideoDevice> pVideoDevice;
+  ComPtr<ID3D11VideoProcessorEnumerator> pEnumerator;
+  ComPtr<ID3D11VideoProcessorEnumerator1> pEnumerator1;
+  ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
+
+  if (FAILED(pD3DDevice.As(&pVideoDevice)))
+  {
+    CLog::LogF(LOGDEBUG, "failed to get video device.");
+    return false;
+  }
+
+  D3D11_VIDEO_PROCESSOR_CONTENT_DESC desc = {};
+  desc.InputWidth = 3840;
+  desc.InputHeight = 2160;
+  desc.OutputWidth = 3840;
+  desc.OutputHeight = 2160;
+
+  if (FAILED(pVideoDevice->CreateVideoProcessorEnumerator(&desc, &pEnumerator)))
+  {
+    CLog::LogF(LOGDEBUG, "failed to create Video Enumerator.");
+    return false;
+  }
+
+  if (FAILED(pEnumerator.As(&pEnumerator1)))
+  {
+    CLog::LogF(LOGDEBUG, "failed to create Video Enumerator1.");
+    return false;
+  }
+
+  const DXGI_FORMAT destFormat = DX::Windowing()->GetBackBuffer().GetFormat();
+  const DXGI_COLOR_SPACE_TYPE destColor = DX::Windowing()->UseLimitedColor()
+                                              ? DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P709
+                                              : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+  BOOL supported = 0;
+
+  // Check if BT.2020 (LEFT) input color space is supported by video driver
+  HRESULT hr = pEnumerator1->CheckVideoProcessorFormatConversion(
+      DXGI_FORMAT_P010, DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P2020, destFormat, destColor,
+      &supported);
+
+  if (SUCCEEDED(hr) && !!supported)
```

Comment:
This seems to be in a few places
```suggestion
  if (SUCCEEDED(hr) && static_cast<bool>(supported))
```

Reply:
I'm curious about this as well. In the end the result is the same, but I wonder what is "more correct", your original, the suggested cast, or something like (supported != 0)

Reply:
Think about type correctness. Any explicit change in "interpretation" of the variable should use a C++ cast.

supported != 0 would be appropriate if `supported` and `0` have the same type on either side of the `=`.